packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2149,7 +2149,7 @@ abstract class RenderBox extends RenderObject {
     assert(!_debugDoingBaseline, 'Please see the documentation for computeDistanceToActualBaseline for the required calling conventions of this method.');
     assert(!debugNeedsLayout);
     assert(() {
-      final RenderObject? parent = this.parent as RenderObject?;
+      final RenderObject? parent = this.parent;
```

Comment:
if we could make this change, we could also make this `this.parent!` and avoid the `!`s below, fwiw

Reply:
oh never mind i'm wrong.